runtime/tests/interpreter/attachments_test.go

```diff
@@ -1599,24 +1599,37 @@ func TestInterpretAttachmentResourceReferenceInvalidation(t *testing.T) {
 		inter, _ := testAccount(t, address, true, `
             resource R {}
             attachment A for R {
-                fun foo(): Int { return 3 }
+                pub(set) var id: UInt8
+                init() {
+                    self.id = 1
+                }
             }
-            fun test() {
+            fun test(): UInt8 {
                 let r <- create R()
                 let r2 <- attach A() to <-r
                 let a = r2[A]!
-                authAccount.save(<-r2, to: /storage/foo)
-                let i = a.foo()
```

Comment:
There was a small catch with this test. When the resource is moved, the underlying atree value of the previous instance of the resource gets cleared and is used to distinguish between valid vs obsolete resources. Here in this case, since the function `a.foo()` doesn't access the atree value, no error is thrown here, even though the reference is no longer pointing to the correct resource.

I updated this test to access a field `a.id` instead of the function call, so it also accesses the internal atree value. then it started failing. But is now fixed once the resource reference tracking is added.